File: library/src/main/java/com/guerinet/utils/Device.java
/*
 * Copyright 2016 Julien Guerinet
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.guerinet.utils;

import android.app.Activity;
import android.content.Context;
import android.graphics.Point;
import android.os.Build;
import android.view.Display;
import android.view.WindowManager;

/**
 * Helper methods relating to the user's device
 * @author Julien Guerinet
 * @since 2.0.2
 */
public class Device {
    /**
     * @return True if the OS is Marshmallow or higher
     */
    public static boolean isMarshmallow() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
    }

    /**
     * @return True fi the OS is Lollipop or higher
     */
    public static boolean isLollipop() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
    }

    /**
     * Gets the device's default display size
     *
     * @param context App context
     * @return Display size
     */
    private static Point getDisplaySize(Context context) {
        WindowManager windowManager;
        if (context instanceof Activity) {
            windowManager = ((Activity) context).getWindowManager();
        } else {
            windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        }

        Display display = windowManager.getDefaultDisplay();
        Point size = new Point();
        display.getSize(size);
        return size;
    }

    /**
     * @param context App context
     * @return Device screen width, in pixels
     */
    public static int getWidth(Context context) {
        return getDisplaySize(context).x;
    }

    /**
     * @param context App context
     * @return Device screen height, in pixels
     */
    public static int getHeight(Context context) {
        return getDisplaySize(context).y;
    }
}

File: library/src/main/java/com/guerinet/suitcase/utils/Device.kt
/*
 * Copyright 2016-2017 Julien Guerinet
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.guerinet.suitcase.utils

import android.app.Activity
import android.content.Context
import android.graphics.Point
import android.os.Build
import android.view.WindowManager

/**
 * Helper methods relating to the user's device
 * @author Julien Guerinet
 * @since 2.0.0
 */
object Device {

    /**
     * @return True if the OS is Nougat or higher, false otherwise
     */
    @JvmStatic
    fun isAtLeastNougat(): Boolean {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
    }

    /**
     * @return True if the OS is Marshmallow or higher, false otherwise
     */
    @JvmStatic
    fun isAtLeastMarshmallow(): Boolean {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
    }

    /**
     * @return True if the OS is Lollipop or higher, false otherwise
     */
    @JvmStatic
    fun isAtLeastLollipop(): Boolean{
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP
    }

    /**
     * @param context App context
     * @return Device screen width, in pixels
     */
    @JvmStatic
    fun getWidth(context: Context): Int {
        return getDisplaySize(context).x
    }

    /**
     * @param context App context
     * @return Device screen height, in pixels
     */
    @JvmStatic
    fun getHeight(context: Context): Int {
        return getDisplaySize(context).y
    }

    /**
     * @return Display size by using the app [context]
     */
    private fun getDisplaySize(context: Context): Point {
        val windowManager: WindowManager
        if (context is Activity) {
            windowManager = context.windowManager
        } else {
            windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        }

        val display = windowManager.defaultDisplay
        val size = Point()
        display.getSize(size)
        return size
    }
}